scripts/ntuple_generators.rb

```diff
@@ -41,21 +41,35 @@ def get_comment(n, algebraic_structure)
 # algebraic_structure is "monoid", "group", "ring", etc.
 #
 # Example return:
-#   "class Tuple2Monoid[T,U](implicit tmonoid : Monoid[T], umonoid : Monoid[U]) extends Monoid[(T,U)]"
+#   "class Tuple2Monoid[T,U](implicit tmonoid: Monoid[T], umonoid: Monoid[U]) extends Monoid[(T,U)]"
 def get_class_definition(n, algebraic_structure)
   # Example: "T,U"
   type_values_commaed = TYPE_SYMBOLS.first(n).join(", ")
-  "class Tuple#{n}#{algebraic_structure.capitalize}[#{type_values_commaed}](implicit #{get_type_parameters(n, algebraic_structure)}) extends #{algebraic_structure.capitalize}[(#{type_values_commaed})]"
+  extends = ["#{algebraic_structure.capitalize}[(#{type_values_commaed})]"]
```

Comment:
everything except for this change is a formatting thing that removes the extra space before `:`. Strange that the checked-in script didn't generate the checked-in code, by the way.